src/common/regions/utils.ts

```diff
@@ -206,17 +204,27 @@ export function filterUndefinedRegions(regions: any[]): Region[] | [] {
   );
 }
 
+// flattens and filters a map of region-type to regions to a list of valid regions
+export const filterGeolocatedRegions = (
+  geolocatedRegions: GeolocatedRegions | null,
+): Region[] => {
+  if (isNull(geolocatedRegions)) {
+    return [];
+  } else {
+    const regionValues = values(geolocatedRegions);
+    const filteredRegions = filterUndefinedRegions(regionValues);
+    return filteredRegions;
+  }
+};
+
 /**
  * We rank a user's geolocated regions (state, county, metro if applicable) first
  * in the searchbar dropdown menu. This sorts the Region[] accordingly.
  */
 export function getAutocompleteRegionsWithGeolocation(
-  geolocation: GeolocationInfo,
+  geolocatedRegions: GeolocatedRegions,
```

Comment:
we can avoid passing the `countyToZipMap` by figuring out the `geolocatedRegions` at a higher level.